x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_creation/engine_creation.test.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { setMockActions, setMockValues } from '../../../__mocks__';
+
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import { EngineCreation } from './';
+
+describe('EngineCreation', () => {
+  const DEFAULT_VALUES = {
+    name: '',
+    rawName: '',
+    language: 'Universal',
+  };
+
+  const MOCK_ACTIONS = {
+    setRawName: jest.fn(),
+    setLanguage: jest.fn(),
+    submitEngine: jest.fn(),
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    setMockValues(DEFAULT_VALUES);
+    setMockActions(MOCK_ACTIONS);
+  });
+
+  it('renders', () => {
+    const wrapper = shallow(<EngineCreation />);
+    expect(wrapper.find('[data-test-subj="EngineCreation"]')).toHaveLength(1);
+  });
+
+  it('EngineCreationForm calls submitEngine on form submit', () => {
+    const wrapper = shallow(<EngineCreation />);
+    const simulatedEvent = {
+      preventDefault: jest.fn(),
```

Comment:
[optional] If we're defining preventDefault in a referenceable var, we might as well check that it was called at the end of the test, e.g.

```js
expect(simulatedEvent.preventDefault).toHaveBeenCalledTimes(1);
expect(MOCK_ACTIONS.submitEngine).toHaveBeenCalledTimes(1);
```

While I'm here also, `.toHaveBeenCalledTimes(1);` (vs. just `.toHaveBeenCalled()`) is a little more specific than we're doing in other files. I don't feel super strongly either way we go, but it would be nice to be consistent across different files vs writing things slightly differently across the codebase.

Reply:
I like the specificity personally